### PR TITLE
feat(flow): add Flow resource

### PR DESCRIPTION
**Pull request by skyscrapr**
Added support for `Flow` resource.

All acceptance tests are passing.

Test output:

```
Running tool: /opt/homebrew/bin/go test -timeout 30s -run ^TestAccResource_flow$ github.com/prefecthq/terraform-provider-prefect/internal/provider/resources

ok  	github.com/prefecthq/terraform-provider-prefect/internal/provider/resources	4.248s
```

### CODE REVIEW

File: docs/data-sources/account_members.md [hunk 1]
```diff
@@ -17,7 +17,14 @@ Use this data source to obtain user or actor IDs to manage Workspace Access.
...
 ```terraform
-# Query all Members in Account
+terraform {
+  required_providers {
+    prefect = {
+      source = "prefecthq/prefect"
+    }
+  }
+}
+
```
Comment (mitchnielsen) [inline docs/data-sources/account_members.md:27]: The provider specification is implied, looks like we don't specify this in any of the other data source docs:

```
~/code/github.com/prefecthq/terraform-provider-prefect/main
$ rg 'required_providers' docs/data-sources

$
```

We can just remove it - what do you think?

File: examples/data-sources/prefect_account_members/data-source.tf [hunk 2]
```diff
@@ -1,2 +1,9 @@
-# Query all Members in Account
+terraform {
+  required_providers {
+    prefect = {
+      source = "prefecthq/prefect"
+    }
+  }
+}
+
```
Comment (mitchnielsen) [inline examples/data-sources/prefect_account_members/data-source.tf:8]: Same here, the provider config is implied, and we don't specify this in any of the other examples.

File: internal/provider/resources/flow.go [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+		},
+	}
+}
+
+// copyFlowToModel copies an api.Flow to a FlowResourceModel.
+func copyFlowToModel(ctx context.Context, flow *api.Flow, model *FlowResourceModel) diag.Diagnostics {
+	model.ID = types.StringValue(flow.ID.String())
+	model.Created = customtypes.NewTimestampPointerValue(flow.Created)
+	model.Updated = customtypes.NewTimestampPointerValue(flow.Updated)
+	// model.WorkspaceID = customtypes.NewUUIDValue(flow.WorkspaceID)
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:140]:
```suggestion
```

I believe we can remove this.

File: internal/provider/resources/flow.go [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// // ImportState imports the resource into Terraform state.
+// func (r *FlowResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+// 	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+// }
+
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:334]:
```suggestion
```

File: internal/provider/resources/flow.go [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *FlowResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Unsupported
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:283]: Any reason this is marked as unsupported? https://app.prefect.cloud/api/docs#tag/Flows/operation/update_flow_api_accounts__account_id__workspaces__workspace_id__flows__id__patch

File: internal/provider/resources/flow.go [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+	// 	// The error from the API call should take precedence
+	// 	// followed by this custom error if a specific deployment is not returned
+	// 	if err == nil && len(deployments) != 1 {
+	// 		err = fmt.Errorf("a deployment with the handle=%s could not be found", model.Handle.ValueString())
+	// 	}
+
+	// 	if len(deployments) == 1 {
+	// 		deployment = deployments[0]
+	// 	}
+	// }
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:259]: What's the plan for this part of the logic?

File: internal/provider/resources/flow.go [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+	}
+
+	// if model.ID.IsNull() && model.Handle.IsNull() {
+	// 	resp.Diagnostics.AddError(
+	// 		"Both ID and Handle are unset",
+	// 		"This is a bug in the Terraform provider. Please report it to the maintainers.",
+	// 	)
+
+	// 	return
+	// }
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:218]:
```suggestion
```

File: internal/testutils/provider.go [hunk 8]
```diff
@@ -17,7 +17,7 @@ import (
 
 // TestAccPrefix is the prefix set for all resources created via acceptance testing,
 // so that we can easily identify and clean them up in case of flakiness/failures.
-const TestAccPrefix = "terraformacc"
+const TestAccPrefix = "tfacc"
```
Comment (mitchnielsen) [inline internal/testutils/provider.go:20]: Why the change here?

**Review by mitchnielsen — CHANGES_REQUESTED** [review]
Thanks again for the contribution @skyscrapr 🤝 

I did a first pass review - just a couple suggestions and looks like there's a few instances of commented out logic that should be either uncommented or removed.

Let me know if I can clarify anything.

**Comment by skyscrapr** [comment]
Yep. Happy to remove. 
Will update. 

Thanks for suggestion.

File: internal/provider/resources/flow.go (hunk 6 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go:259]: no plan, there is no need for this code. removed.

File: internal/provider/resources/flow.go (hunk 5 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go:283]: The default behaviour would be delete and recreate on a change. 
It's unclear if the use case to update in place adds much value.
Keen to understand if there's requirement.

File: internal/testutils/provider.go (hunk 8 above)
Comment (skyscrapr) [reply internal/testutils/provider.go:20]: there is a limit to the length of this string

**Comment by skyscrapr** [comment]
I had to regenerate docs due to the feedback and I've updated the branch with the suggested changes.

File: internal/provider/resources/flow.go (hunk 5 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go]: My suggestion would be to see if customers need this capability. 
From what i can see the effort to update is not adding much.

File: internal/testutils/provider.go (hunk 8 above)
Comment (skyscrapr) [reply internal/testutils/provider.go:20]: actually i just retried this and it appears to no longer be the case.
Ill revert this change

File: internal/provider/resources/flow.go (hunk 5 above)
Comment (mitchnielsen) [reply internal/provider/resources/flow.go]: @parkedwards - could you weigh in here since you have more context? I'm thinking that if we choose not to support `Update` then let's remove the method entirely to avoid confusion while developing.

Comment (parkedwards) [reply internal/provider/resources/flow.go]: i think @skyscrapr's intuition is correct here. it looks like we only accept changes to the `tags` parameter on the flow resource. practically, we'd probably re-create the flow on the API if we _**did**_ need to make any material change

that said, i do think we need to keep `Update` in place, as we need to implement the `resource.Resource` interface

Comment (skyscrapr) [reply internal/provider/resources/flow.go]: Yes. The method has to remain. 
But the standard way to not support update is to leave this method blank. 

The core will then treat any changes in the config as a destroy/create for the resource.

Comment (mitchnielsen) [reply internal/provider/resources/flow.go]: Got it, thank you both 👍🏼

Comment (mitchnielsen) [reply internal/provider/resources/flow.go]: @skyscrapr - given the result of the discussion here, it sounds to me like we don't want users to be able to update Flows and instead force a replace. We can use PlanModifiers for that. For example:

```diff
diff --git a/internal/provider/resources/flow.go b/internal/provider/resources/flow.go
index d568c2b..669bae3 100644
--- a/internal/provider/resources/flow.go
+++ b/internal/provider/resources/flow.go
@@ -12,6 +12,7 @@ import (
        "github.com/hashicorp/terraform-plugin-framework/resource"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema/listdefault"
+       "github.com/hashicorp/terraform-plugin-framework/resource/schema/listplanmodifier"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
        "github.com/hashicorp/terraform-plugin-framework/types"
@@ -127,6 +128,9 @@ func (r *FlowResource) Schema(_ context.Context, _ resource.SchemaRequest, resp
                                Optional:    true,
                                Computed:    true,
                                Default:     listdefault.StaticValue(defaultEmptyTagList),
+                               PlanModifiers: []planmodifier.List{
+                                       listplanmodifier.RequiresReplace(),
+                               },
                        },
                },
        }
```

Without this change, when I try to update an existing Flow resource with an additional tag, I get:

```
Terraform will perform the following actions:

  # prefect_flow.flow will be updated in-place
  ~ resource "prefect_flow" "flow" {
      ~ created = "2024-06-13T19:29:06Z" -> (known after apply)
        id      = "54e09681-9535-4f0b-9a15-ffd999c7187e"
        name    = "test-flow"
      ~ tags    = [
            "test",
          + "test2",
        ]
      ~ updated = "2024-06-13T19:29:06Z" -> (known after apply)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

prefect_flow.flow: Modifying... [id=54e09681-9535-4f0b-9a15-ffd999c7187e]
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to prefect_flow.flow, provider "provider[\"registry.terraform.io/prefecthq/prefect\"]" produced an unexpected new value:
│ .tags: element 1 has vanished.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

_With_ the diff applied using the PlanModifier, I now get something more helpful:

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # prefect_flow.flow must be replaced
-/+ resource "prefect_flow" "flow" {
      ~ created = "2024-06-13T19:29:06Z" -> (known after apply)
      ~ id      = "54e09681-9535-4f0b-9a15-ffd999c7187e" -> (known after apply)
        name    = "test-flow"
      ~ tags    = [ # forces replacement
            "test",
          + "test2",
        ]
      ~ updated = "2024-06-13T19:29:06Z" -> (known after apply)
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

prefect_flow.flow: Destroying... [id=54e09681-9535-4f0b-9a15-ffd999c7187e]
prefect_flow.flow: Destruction complete after 0s
prefect_flow.flow: Creating...
prefect_flow.flow: Creation complete after 0s [id=c153db34-3aa9-4551-b07b-7e58df3ac9e6]

Apply complete! Resources: 1 added, 0 changed, 1 destroyed
```

What do you think?

Comment (skyscrapr) [reply internal/provider/resources/flow.go]: Looks good to me. I added a test to check for updating tags in the flow resource. I had to add the plan modifier suggested to get the correct functionality. tests pass so confirms that changing tags will work correctly.

good suggestion thanks!

File: internal/provider/resources/flow.go [hunk 9]
```diff
@@ -0,0 +1,355 @@
...
+	client, err := r.client.Flows(model.AccountID.ValueUUID(), model.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating deployment client",
+			fmt.Sprintf("Could not create deployment client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
+		)
+	}
+
+	// A deployment can be imported + read by either ID or Handle
+	// If both are set, we prefer the ID
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:231]: based on the logic below it looks like we're only importing by ID for now, right?

File: internal/provider/resources/flow.go [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *FlowResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Unsupported - redeploy by default
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *FlowResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var model FlowResourceModel
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:276]: Let's change this variable name to `state` per my earlier comment.

File: internal/provider/resources/flow.go [hunk 11]
```diff
@@ -0,0 +1,355 @@
...
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *FlowResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var model FlowResourceModel
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:205]: Let's change this variable name to `state` per my earlier comment.

File: internal/provider/resources/flow.go [hunk 12]
```diff
@@ -0,0 +1,355 @@
...
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// ImportState imports the resource into Terraform state.
+func (r *FlowResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// we'll allow input values in the form of:
+	// - "workspace_id,name"
+	// - "name"
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:322]: Could you please add a new directory showing examples for importing Flows?

For example, you can reference [examples/resources/prefect_variable/](https://github.com/skyscrapr/terraform-provider-prefect/tree/resource_flow/examples/resources/prefect_variable).

File: internal/client/flows.go [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+	"net/http"
+
+	"github.com/google/uuid"
+
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+)
+
+var _ = api.FlowsClient(&FlowsClient{})
+
+// FlowsClient is a client for working with Deployments.
```
Comment (mitchnielsen) [inline internal/client/flows.go:18]:
```suggestion
// FlowsClient is a client for working with Flows.
```

File: internal/provider/resources/flow.go [hunk 14]
```diff
@@ -0,0 +1,355 @@
...
+	// 	)
+
+	// 	return
+	// }
+
+	client, err := r.client.Flows(model.AccountID.ValueUUID(), model.WorkspaceID.ValueUUID())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating deployment client",
+			fmt.Sprintf("Could not create deployment client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:226]:
```suggestion
			"Error creating flow client",
			fmt.Sprintf("Could not create flow client, unexpected error: %s. This is a bug in the provider, please report this to the maintainers.", err.Error()),
```

File: internal/provider/resources/flow.go [hunk 15]
```diff
@@ -0,0 +1,355 @@
...
+		)
+	}
+
+	// A deployment can be imported + read by either ID or Handle
```
Comment (mitchnielsen) [inline internal/provider/resources/flow.go:230]:
```suggestion
	// A flow can be imported + read by either ID or Handle
```

File: internal/provider/resources/flow_test.go [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+	resourceName := "prefect_flow.flow"
+	workspaceResourceName := "prefect_workspace.workspace"
+	randomName := testutils.TestAccPrefix + acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProtoV6ProviderFactories: testutils.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { testutils.AccTestPreCheck(t) },
+		Steps: []resource.TestStep{
+			{
+				// Check creation + existence of the deployment resource
```
Comment (mitchnielsen) [inline internal/provider/resources/flow_test.go:39]:
```suggestion
				// Check creation + existence of the flow resource
```

File: internal/provider/resources/flow.go (hunk 9 above)
Comment (mitchnielsen) [reply internal/provider/resources/flow.go:231]: Currently, when I try to import a flow, I get:

```
$ tf import prefect_flow.flow 85f06a57-d187-416e-b49a-8beb307b9afb
prefect_flow.flow: Importing from ID "85f06a57-d187-416e-b49a-8beb307b9afb"...
prefect_flow.flow: Import prepared!
  Prepared prefect_flow for import
prefect_flow.flow: Refreshing state...
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-prefect plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105491a50]

goroutine 66 [running]:
github.com/prefecthq/terraform-provider-prefect/internal/provider/resources.copyFlowToModel({0x1057aa0e8, 0x14000489560}, 0x0, 0x140000ee3c0)
        /Users/mitch/code/github.com/prefecthq/terraform-provider-prefect/resource_flow/internal/provider/resources/flow.go:141 +0x30
github.com/prefecthq/terraform-provider-prefect/internal/provider/resources.(*FlowResource).Read(0x1400048ae50, {0x1057aa0e8, 0x14000489560}, {{{{0x1057af758, 0x140005160f0}, {0x10569c100, 0x14000516030}}, {0x1057b11d8, 0x14000433040}}, 0x140004440c0, ...}, ...)
        /Users/mitch/code/github.com/prefecthq/terraform-provider-prefect/resource_flow/internal/provider/resources/flow.go:258 +0x39c
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0x1400018d340, {0x1057aa0e8, 0x14000489560}, 0x140004895f0, 0x14000297690)
        /Users/mitch/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/fwserver/server_readresource.go:101 +0x510
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0x1400018d340, {0x1057aa0e8?, 0x14000488e70?}, 0x140004a2600)
        /Users/mitch/go/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/proto6server/server_readresource.go:55 +0x2e0
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0x1400038e000, {0x1057aa0e8?, 0x140004884e0?}, 0x14000530060)
        /Users/mitch/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/tf6server/server.go:776 +0x3c8
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x105778520, 0x1400038e000}, {0x1057aa0e8, 0x140004884e0}, 0x140001a4800, 0x0)
        /Users/mitch/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:482 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001a1000, {0x1057aa0e8, 0x140004883f0}, {0x1057b0060, 0x140000bf860}, 0x1400021e000, 0x14000383050, 0x105d26688, 0x0)
        /Users/mitch/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1386 +0xb58
google.golang.org/grpc.(*Server).handleStream(0x140001a1000, {0x1057b0060, 0x140000bf860}, 0x1400021e000)
        /Users/mitch/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1797 +0xb10
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /Users/mitch/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1027 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 13
        /Users/mitch/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1038 +0x13c

Error: The terraform-provider-prefect plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Review by mitchnielsen — CHANGES_REQUESTED** [review]
Thanks for the changes @skyscrapr 🤝 

I did another pass through the code along with some testing and have some more small requests.

File: internal/provider/resources/flow.go (hunk 9 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go:230]: hmm that's odd. the acceptance test for import passes fine. Let me take a closer look at this one.

Comment (skyscrapr) [reply internal/provider/resources/flow.go:230]: ok, some investigation into this. Import is working for the following scenario:
`terraform import prefect_flow.flow workspace_id,flow_id`
But in other scenarios it panics.
I'm looking into it.

File: internal/provider/resources/flow.go [hunk 17]
```diff
@@ -0,0 +1,355 @@
...
+		return diags
+	}
+	model.Tags = tags
+
+	return nil
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *FlowResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var model FlowResourceModel
```
[the remark this answers is not in the document]
Comment (skyscrapr) [reply internal/provider/resources/flow.go:156]: done

File: internal/provider/resources/flow.go (hunk 7 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go]: done

File: internal/provider/resources/flow.go (hunk 9 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go]: ok, I made a couple of updates to support import correctly using ids. 
It's possible to do this via name too. 
But suggest we wait to see if any users request this as it meets the basic requirements currently.
Please take another look.

File: internal/provider/resources/flow.go (hunk 12 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go]: changed to support ids only.

File: internal/provider/resources/flow.go (hunk 11 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go:204]: done

File: internal/provider/resources/flow.go (hunk 9 above)
Comment (mitchnielsen) [reply internal/provider/resources/flow.go]: Really appreciate the iteration here, thanks @skyscrapr.

Looks like we just need to better handle the scenario where the `inputParts` only equals `1`. You can see [block's import logic as an example](https://github.com/PrefectHQ/terraform-provider-prefect/blob/5ff52364630ea269fc336c8e66601a3cb381ad26/internal/provider/resources/block.go). I was able to copy some of that logic and get `tf import` working:

```diff
diff --git a/internal/provider/resources/flow.go b/internal/provider/resources/flow.go
index d26e379..97c609d 100644
--- a/internal/provider/resources/flow.go
+++ b/internal/provider/resources/flow.go
@@ -333,10 +333,17 @@ func (r *FlowResource) ImportState(ctx context.Context, req resource.ImportState
 		return
 	}
 
-	id := inputParts[maxInputCount-1]
-	if len(inputParts) == maxInputCount {
-		workspaceID := inputParts[0]
-		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("workspace_id"), workspaceID)...)
+	identifier := inputParts[0]
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), identifier)...)
+
+	if len(inputParts) == 2 && inputParts[1] != "" {
+		workspaceID, err := uuid.Parse(inputParts[1])
+		if err != nil {
+			resp.Diagnostics.AddError("problem parsing workspace ID", "see details")
+
+			return
+		}
+
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("workspace_id"), workspaceID.String())...)
 	}
-	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), id)...)
 }
```

```shell
$ tf import prefect_flow.flow c9bb05e4-2efe-49f7-b4e2-318d295553b6
prefect_flow.flow: Importing from ID "c9bb05e4-2efe-49f7-b4e2-318d295553b6"...
prefect_flow.flow: Import prepared!
  Prepared prefect_flow for import
prefect_flow.flow: Refreshing state... [id=c9bb05e4-2efe-49f7-b4e2-318d295553b6]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

>  But suggest we wait to see if any users request this as it meets the basic requirements currently.

I'm not opposed to that, but let's at least confirm that diagnostics properly return this information to communicate that workspace ID currently needs to be supplied (meaning this won't work on OSS for now). Currently, `id := inputParts[maxInputCount-1]` throws an error via stack trace rather than diagnostics. So let's either get that method working based of my example above, or throw correct diagnostics for now 👍🏼

Comment (skyscrapr) [reply internal/provider/resources/flow.go]: ok, applied suggested changes.

File: internal/provider/resources/flow.go (hunk 12 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go]: updated as suggested

**Review by mitchnielsen — APPROVED** [review]
I'll be out for the next week and a half so I'll hand this over to @parkedwards for a final look.

Thanks again for the iteration on this one @skyscrapr 🤝 

Using the config below, I was able to create, update (replace), import, and destroy a Flow. Looking good from my side ✅ 

```terraform
terraform {
  required_providers {
    prefect = {
      source = "registry.terraform.io/prefecthq/prefect"
    }
  }
}

provider "prefect" {
  endpoint = "http://localhost:4200/api"
}

resource "prefect_flow" "flow" {
  name = "test-flow"
  tags = ["test", "test2", "test3"]
}
```

**Comment by skyscrapr** [comment]
Awesome. Thanks for the thorough review. Once this is merged I'll update #202 to the same level. 

Cheers

File: internal/provider/resources/flow.go [hunk 18]
```diff
@@ -0,0 +1,347 @@
...
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// ImportState imports the resource into Terraform state.
+func (r *FlowResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// we'll allow input values in the form of:
+	// - "workspace_id,id"
```
Comment (parkedwards) [inline internal/provider/resources/flow.go:311]:
```suggestion
	// - "id,workspace_id"
```

File: internal/provider/resources/flow.go [hunk 19]
```diff
@@ -0,0 +1,347 @@
...
+	// - "id"
+	maxInputCount := 2
+	inputParts := strings.Split(req.ID, ",")
+
+	// eg. "foo,bar,baz"
+	if len(inputParts) > maxInputCount {
+		resp.Diagnostics.AddError(
+			"Unexpected Import Identifier",
+			fmt.Sprintf("Expected a maximum of 2 import identifiers, in the form of `workspace_id,name`. Got %q", req.ID),
```
Comment (parkedwards) [inline internal/provider/resources/flow.go:320]:
```suggestion
			fmt.Sprintf("Expected a maximum of 2 import identifiers, in the form of `name,workspace_id`. Got %q", req.ID),
```

File: internal/provider/resources/flow.go [hunk 20]
```diff
@@ -0,0 +1,347 @@
...
+				// https://github.com/hashicorp/terraform-plugin-framework/issues/754
+				Description: "Flow ID (UUID)",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"created": schema.StringAttribute{
+				Computed:    true,
+				CustomType:  customtypes.TimestampType{},
+				Description: "Timestamp of when the resource was created (RFC3339)",
```
Comment (parkedwards) [inline internal/provider/resources/flow.go:102]: can we use a a `stringplanmodifier.UseStateForUnknown()` planmodifier for `created` here?

File: internal/provider/helpers/import.go [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+)
+
+func GetResourceWorkspaceImportStateID(resourceName string, workspaceName string) resource.ImportStateIdFunc {
```
Comment (parkedwards) [inline internal/provider/helpers/import.go:11]: nice

File: internal/provider/resources/flow.go (hunk 20 above)
Comment (skyscrapr) [reply internal/provider/resources/flow.go:102]: Interesting suggestion. Let me take a look at that to see what the experience is.

Comment (skyscrapr) [reply internal/provider/resources/flow.go:102]: since flows currently don't support updating, I'm not sure this adds anything to the overall experience. Happy either way.

Comment (parkedwards) [reply internal/provider/resources/flow.go:102]: ah that's right, they'll be re-created every time. we can disregard

**Review by parkedwards — APPROVED** [review]
LGTM - thank you!
